native-oci-generative-ai/README.md

```diff
@@ -0,0 +1,56 @@
+# OCI Native Generative AI Example
+
+This example illustrates how to use the Generative AI service provided by Oracle Cloud Infrastructure (OCI). It uses a state-of-the-art, customizable large language model to generate text. 
+
+The [source code](./src/main/java/com/oracle/labs/GenerateTextExample.java) is a self-contained demonstration of using a [GenerativeAiInferenceClient](https://docs.oracle.com/iaas/tools/java/latest/com/oracle/bmc/generativeaiinference/GenerativeAiInferenceClient.html) to generate text, based on a prompt.
+
+The [_pom.xml_](pom.xml) file includes necessary dependents (such as the [OCI Java SDK](https://docs.public.oneportal.content.oci.oraclecloud.com/iaas/Content/API/SDKDocs/javasdk.htm)) and configures the build, including the [Native Maven Plugin](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html) to generate a native executable with GraalVM Native Image.
+
+> For more information, see [Generative AI](https://docs.oracle.com/iaas/Content/generative-ai/home.htm).
+
+## Prerequisites
+* [GraalVM](https://www.graalvm.org/downloads/).
+
+* You must have access to access to Generative AI resources in your tenancy/compartment. (For more information, see [Getting Access to Generative AI](https://docs.oracle.com/iaas/Content/generative-ai/iam-policies.htm).)
+
+## Preparing Your Code
+
+Download or clone the GraalVM demos repository:
+```sh
+git clone https://github.com/graalvm/graalvm-demos
+```
+
+Edit the contents of [_GenerateTextExample.java_](./src/main/java/com/oracle/labs/GenerateTextExample.java) as follows:
+
+1. Update the value of the static variable `CONFIG_PROFILE` with the name of your OCI profile name.
```

Comment:
These could just be Java properties or environment variables, which can then be set at run-time (or frozen into the executable via build-time init if you insist)?